### DISCUSSION TITLE
Need help getting started with metrics

### DISCUSSION THREAD

**Discussion by cstar** [open-telemetry/opentelemetry-erlang#758] [Q&A]
Hi,

I'd like to use opentelemetry metrics and logs with my Erlang app, but I am really struggling to get started.
I'm using Otel collector to receive metrics, and I see it collects spans from my app, but the following code does not produce anything:

```erlang
-module(metrics_test).

-include_lib("opentelemetry_api/include/opentelemetry.hrl").
-include_lib("opentelemetry_api_experimental/include/otel_meter.hrl").
-export([start/0]).

start() ->
    application:ensure_all_started(opentelemetry),
    application:ensure_all_started(opentelemetry_experimental),
    application:ensure_all_started(tls_certificate_check),
    application:ensure_all_started(ssl_verify_fun),
    CounterName = m_counter,
    CounterDesc = <<"macro made counter description">>,
    CounterUnit = kb,

    ?create_counter(CounterName, #{description => CounterDesc,
                                             unit => CounterUnit}),

    ?counter_add(CounterName, 5).
``` 

my sys.config:
```erlang
[
  {otel_getting_started, []},
  {opentelemetry,[
    {span_processor, batch},
    {traces_exporter, otlp},
    {metrics_exporter, otlp}
  ]},
  {
    opentelemetry_exporter,[
      {otlp_protocol, http_protobuf},
      {otlp_endpoint, "http://localhost:4318"}]
  }
].
```

What am I missing?

Thanks,

**Comment by tsloughter** [comment]
Could you provide a sample repo I can clone and try out?

**Reply by cstar** [reply]
Hi @tsloughter 

Here it is: https://github.com/cstar/otel_metrics_test

In the `node/` directory, a bit of nodejs to check the collector collects appropriately.
Thanks!

**Reply by tsloughter** [reply]
Thanks, I see the issue and should have caught it without having to ask you to do that :).

`metrics_exporter` isn't used. That needs to either be removed from `otel_configuration` or made to works somehow, it looks like most just don't support this configuration option bc it doesn't fit well into the model of metrics exporting, https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md

Currently what you need to do is configure a "Reader" with an exporter configured inside it. Try with:

```
[
  {otel_getting_started, []},
  {opentelemetry,[
    {span_processor, batch},
    {traces_exporter, otlp},
  ]},


 {opentelemetry_experimental,
  [{readers, [#{module => otel_metric_reader,
                config => #{export_interval_ms => 1000,
                            exporter => {opentelemetry_exporter, #{}}}}]}]},

 {
  opentelemetry_exporter,[
                          {otlp_protocol, http_protobuf},
                          {otlp_endpoint, "http://localhost:4318"}]
 }
].
```

Also see the example here, https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/examples/roll_dice_elli/config/sys.config

**Comment by cstar** [comment]
Hi Tristan, I still have issues — I've pinned Otel Erlang to 1.4, but I am still not able to publish metrics to the Otel Collector.
Please see updated code in the repo:
https://github.com/cstar/otel_metrics_test
What am I missing?
Best,
